### PR TITLE
Feature/fate318211 clusterscripts

### PR DESCRIPTION
**Pull request by tomschr**
Implemented FATE 318211

### CODE REVIEW

File: xml/ha_config_cli.xml [hunk 1]
```diff
@@ -191,22 +194,34 @@
        agents).
       </para>
      </formalpara>
+      <para>It is possible to abbreviate subcommands as long as they are
+        unique. For example, you can shorten <command>status</command> as
+      <command>st</command> and &crmsh; will know what you have meant.
+      </para>
+      <para>Another feature is to shorten parameters. Usually, you add
+        parameters through the <command>params</command> keyword.
+        You can leave out the params section if it the first and only section.
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: missing word? if it *is* the first and only....

File: xml/ha_config_cli.xml [hunk 2]
```diff
@@ -191,22 +194,34 @@
...
        Use <command>crm</command> and its subcommands in a script. This can
        be done in two ways:
       </para>
      </formalpara>
-<screen>&prompt.root;<command>crm</command> -f script.cli
-&prompt.root;<command>crm</command> &lt; script.cli</screen>
+<screen>&prompt.root;<command>crm</command> -f example.cli
+&prompt.root;<command>crm</command> &lt; example.cli</screen>
      <para>
       The script can contain any command from <command>crm</command>. For
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: any command from <command>crm</command> -> any <command>crm</command> command ?

File: xml/ha_config_cli.xml (hunk 1 above)
Comment (tomschr) [inline xml/ha_config_cli.xml]: Right, fixed.

File: xml/ha_config_cli.xml (hunk 2 above)
Comment (tomschr) [inline xml/ha_config_cli.xml]: Hmn... actually I wanted to say is that the script can contain **any subcommand** of crm. Is that better?

File: xml/ha_config_cli.xml [hunk 3]
```diff
@@ -217,8 +232,24 @@
      </para>
     </listitem>
     <listitem>
+      <formalpara>
+        <title>As &crmsh; Cluster Scripts:</title>
+        <para>These are a collection of metadata, references to RPM packages,
+          configuration files, and &crmsh; subommands bundled under a single,
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: typo: sucommands -> subcommands
These are managed -> They are managed (to avoid double "these")

File: xml/ha_config_cli.xml [hunk 4]
```diff
@@ -217,8 +232,24 @@
...
+          yet descriptive name. These are managed through the
+          <command>crm script</command> command.
+        </para>
+      </formalpara>
+      <para>Do not confuse them with crm shell scripts: although both share
+        some common objectives, the crm shell scripts just contains subcommands
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: typo: contains - > contain

File: xml/ha_config_cli.xml [hunk 5]
```diff
@@ -217,8 +232,24 @@
...
+        whereas cluster scripts incorporate much more and goes beyond simple
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: typo: goes -> go 
I'm not sure about the "goes beyond" part -> maybe remove it?

File: xml/ha_config_cli.xml [hunk 6]
```diff
@@ -356,9 +387,187 @@ Operations' defaults (advisory minimum):
...
+      <listitem>
+        <para>Collect information and report potential problems with the
+          cluster.</para>
+      </listitem>
+      <listitem>
+        <para>Deploy the changes to all nodes.</para>
+      </listitem>
+    </itemizedlist>
+
+    <para>It is not intented to replace other tools for managing clusters,
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: Maybe rephrase to: &cmrsh; cluster scripts do not replace other tools for managing clusters&mdash; they provide an integrated way to perform the above tasks across the cluster.

File: xml/ha_config_cli.xml [hunk 7]
```diff
@@ -356,9 +387,187 @@ Operations' defaults (advisory minimum):
...
+    <title>Using &crmsh;'s Cluster Scripts</title>
+    <remark>toms 2015-10-14: FATE#318211</remark>
+    <para>One big task of configuring and managing a cluster is to collect
+      all information of its nodes and deploying the changes. Performing the
+      same task manually on different nodes can be error-prone. To make this
+      process simpler, easier, repetitive, and more consistent, the &crmsh; cluster
+      scripts were invented.
+    </para>
+
+    <para>The &crmsh; cluster scripts have the following objectives:</para>
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: I would use the "-ing" form for all of the listitems

File: xml/ha_config_cli.xml (hunk 3 above)
Comment (tomschr) [inline xml/ha_config_cli.xml]: Ok, fixed.

File: xml/ha_config_cli.xml (hunk 4 above)
Comment (tomschr) [inline xml/ha_config_cli.xml]: Fixed.

File: xml/ha_config_cli.xml [hunk 8]
```diff
@@ -356,9 +387,187 @@ Operations' defaults (advisory minimum):
    </tip>
   </sect2>
 
+  <sect2 xml:id="sec.ha.manual_config.clusterscripts">
+    <title>Using &crmsh;'s Cluster Scripts</title>
+    <remark>toms 2015-10-14: FATE#318211</remark>
+    <para>One big task of configuring and managing a cluster is to collect
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: Perhaps shorten this para to: Collecting information from all cluster nodes and deploying any changes is a key cluster administration task. Instead of performing the same procedures manually on different nodes (which is error-prone), you can use the &crmsh; cluster scripts.

File: xml/ha_config_cli.xml (hunk 5 above)
Comment (tomschr) [inline xml/ha_config_cli.xml]: Yes. Changed it to:

```
whereas cluster scripts incorporate much more than a simple enumeration of commands.
```

File: xml/ha_config_cli.xml (hunk 6 above)
Comment (tomschr) [inline xml/ha_config_cli.xml]: Used your phrase. Thanks! :+1:

File: xml/ha_config_cli.xml (hunk 7 above)
Comment (tomschr) [inline xml/ha_config_cli.xml]: Right, added -ing form.

File: xml/ha_config_cli.xml (hunk 8 above)
Comment (tomschr) [inline xml/ha_config_cli.xml]: Great, much better!

File: xml/ha_config_cli.xml [hunk 9]
```diff
@@ -356,9 +387,187 @@ Operations' defaults (advisory minimum):
...
+        xlink:href="http://crmsh.github.io/scripts/"
+        xlink:show="new"/>.
+    </para>
+
+    <sect3 xml:id="sec.ha.manual_config.clusterscripts.usage">
+      <title>Usage</title>
+      <para>To get a list of all available cluster scripts, run:</para>
+      <screen>&prompt.root;<command>crm</command> script list</screen>
+
+      <para>Each cluster script contains more information than just a
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: Maybe shorten to: To view the components of a script, use the <command>show</command> command and the name of the cluster script, for example:

File: xml/ha_config_cli.xml [hunk 10]
```diff
@@ -356,9 +387,187 @@ Operations' defaults (advisory minimum):
...
+            </row>
+            <row>
+              <entry><parameter>nodes</parameter></entry>
+              <entry><replaceable>LIST</replaceable></entry>
+              <entry>List of nodes to execute the script for</entry>
+            </row>
+            <row>
+              <entry><parameter>port</parameter></entry>
+              <entry><replaceable>NUMBER</replaceable></entry>
+              <entry>Port to connect on</entry>
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: to connect *to*

File: xml/ha_config_cli.xml [hunk 11]
```diff
@@ -356,9 +387,187 @@ Operations' defaults (advisory minimum):
...
+              <entry><parameter>statefile</parameter></entry>
+              <entry><replaceable>FILE</replaceable></entry>
+              <entry>When single-stepping, the state is saved in the given
+                file </entry>
+            </row>
+            <row>
+              <entry><parameter>sudo</parameter></entry>
+              <entry><replaceable>BOOL</replaceable></entry>
+              <entry>If set, crm will prompt for a sudo password and use sudo
+                when appropriate (default: no) </entry>
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: when -> where

File: xml/ha_config_cli.xml [hunk 12]
```diff
@@ -356,9 +387,187 @@ Operations' defaults (advisory minimum):
...
+    </sect3>
+
+    <sect3 xml:id="sec.ha.manual_config.clusterscripts.verify_run">
+      <title>Verifying and Running a Cluster Script</title>
+      <para>Before you actually run a cluster script it is advisable to
+          review the actions that it will perform. A cluster script can
+          potentially perform a series of actions and may fail for
+          various reasons. Thus, verifying your parameters before
+          running it helps to avoid problems.</para>
+      <para>For example, the mailto resource agent requires an unique identifier
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: mailto -> tag with <literal/> or <systemitem/>?

File: xml/ha_config_cli.xml (hunk 9 above)
Comment (tomschr) [inline xml/ha_config_cli.xml]: Great! Used your sentence.

File: xml/ha_config_cli.xml (hunk 10 above)
Comment (tomschr) [inline xml/ha_config_cli.xml]: Fixed

File: xml/ha_config_cli.xml (hunk 11 above)
Comment (tomschr) [inline xml/ha_config_cli.xml]: Fixed

File: xml/ha_config_cli.xml (hunk 12 above)
Comment (tomschr) [inline xml/ha_config_cli.xml]: Used \<systemitem>

File: xml/ha_config_cli.xml [hunk 13]
```diff
@@ -356,9 +387,187 @@ Operations' defaults (advisory minimum):
...
+              <entry>Run script as the given user </entry>
+            </row>
+          </tbody>
+        </tgroup>
+      </table>
+    </sect3>
+
+    <sect3 xml:id="sec.ha.manual_config.clusterscripts.verify_run">
+      <title>Verifying and Running a Cluster Script</title>
+      <para>Before you actually run a cluster script it is advisable to
```
Comment (taroth21) [inline xml/ha_config_cli.xml]: Perhaps shorten to: Before running a cluster script review the actions that it will perform and verify its parameters to avoid problems.

**Comment by tomschr** [comment]
Merged https://github.com/SUSE/doc-sleha/commit/5ea3b5c519309df32a5406cb227759fcbe7561df
